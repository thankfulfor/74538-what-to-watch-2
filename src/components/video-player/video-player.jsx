import React from 'react';
import PropTypes from 'prop-types';

export class VideoPlayer extends React.PureComponent {
  constructor(props) {
    super(props);

    this.videoRef = React.createRef();
    this.muted = true;
  }

  componentDidMount() {
    const videoElement = this.videoRef.current;
    const {isPlaying} = this.props;

    if (isPlaying) {
      videoElement.play();
      this.muted = true;
    }
  }

  componentDidUpdate() {
    const videoElement = this.videoRef.current;
    const {isPlaying} = this.props;

    return isPlaying ? videoElement.play() : videoElement.pause();
  }

  render() {
    const {src, poster} = this.props;

    return (
      <video
        ref={this.videoRef}
        src={src}
        poster={poster}
        style={{width: `100%`, height: `100%`}}
        muted={this.muted}
      />
    );
  }
}

VideoPlayer.propTypes = {
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired,
  isPlaying: PropTypes.bool.isRequired,
};
