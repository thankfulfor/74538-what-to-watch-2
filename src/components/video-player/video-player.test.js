import React from 'react';
import renderer from 'react-test-renderer';
import {VideoPlayer} from './video-player.jsx';

const film = {
  previewImage: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv`
};

it(`VideoPlayer корректно рендерится после перезапуска`, () => {
  const tree = renderer
    .create(
        <VideoPlayer
          src={film.previewVideoLink}
          poster={film.previewImage}
          isPlaying={false}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
