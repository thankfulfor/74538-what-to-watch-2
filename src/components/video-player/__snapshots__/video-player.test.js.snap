// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer корректно рендерится после перезапуска 1`] = `
<video
  poster="img/we-need-to-talk-about-kevin.jpg"
  src="https://upload.wikimedia.org/wikipedia/commons/b/b3/Big_Buck_Bunny_Trailer_400p.ogv"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
/>
`;
